File: MajorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int> m;
        for(auto num : nums){
            if(m.find(num) == m.end())
                m[num] = 1;
            else
                m[num]++;
            if(m[num] > (nums.size()/2))
                return num;
        }
        return 0;
        
    }
};

File: Majority Element.py
# Day: 6

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        dic = {}
        for num in nums:
            dic[num] = dic.get(num, 0) + 1
        for num in nums:
            if dic[num] > len(nums)//2:
                return num
